Mathlib/Topology/KrullDimension.lean

```diff
@@ -62,3 +74,123 @@ theorem IsHomeomorph.topologicalKrullDim_eq (f : X → Y) (h : IsHomeomorph f) :
     IsClosedEmbedding.topologicalKrullDim_le (h.homeomorph f).symm
     (h.homeomorph f).symm.isClosedEmbedding
   le_antisymm fwd bwd
+
+/-! ### Maps between irreducible closed sets -/
+
+/-- The canonical map from irreducible closed sets of a subspace `Y` to irreducible
+closed sets of the ambient space `X`, defined by taking the closure of the image
+under the inclusion map. This map is crucial for comparing Krull dimensions. -/
+def mapIrreducibleClosed (Y : Set X) (c : IrreducibleCloseds Y) : IrreducibleCloseds X where
+  carrier := closure (Subtype.val '' c.carrier)
+  is_irreducible' := c.is_irreducible'.image Subtype.val
+    (continuous_subtype_val.continuousOn) |>.closure
+  is_closed' := isClosed_closure
+
+/-- The map `mapIrreducibleClosed` is injective, meaning distinct irreducible
+closed sets in a subspace map to distinct irreducible closed sets in the ambient space.
+This ensures that the dimension-preserving properties hold. -/
+lemma mapIrreducibleClosed_injective (Y : Set X) :
+    Function.Injective (mapIrreducibleClosed Y) := by
+  intro A B h_images_eq
+  ext x
+  have h_closures_eq : closure (Subtype.val '' A.carrier) =
+      closure (Subtype.val '' B.carrier) :=
+    congrArg IrreducibleCloseds.carrier h_images_eq
+  constructor
+  · -- Forward direction: x ∈ A → x ∈ B
+    intro hx_in_A
+    change x ∈ B.carrier
+    rw [← B.is_closed'.closure_eq]
+    -- Use the mathlib lemma for embeddings
+    rw [Topology.IsEmbedding.subtypeVal.closure_eq_preimage_closure_image]
+    rw [← h_closures_eq]
+    simp_rw [mem_preimage]
+    apply subset_closure
+    exact mem_image_of_mem Subtype.val hx_in_A
+  · -- Backward direction: x ∈ B → x ∈ A
+    intro hx_in_B
+    change x ∈ A.carrier
+    rw [← A.is_closed'.closure_eq]
+    -- Use the mathlib lemma for embeddings
+    rw [Topology.IsEmbedding.subtypeVal.closure_eq_preimage_closure_image]
+    rw [h_closures_eq]
+    simp_rw [mem_preimage]
+    apply subset_closure
+    exact mem_image_of_mem Subtype.val hx_in_B
+
+
+/-! ### Partial order structure on irreducible closed sets -/
+
+instance : PartialOrder (IrreducibleCloseds X) where
+  le s t := s.carrier ⊆ t.carrier
+  le_refl s := Set.Subset.refl _
+  le_trans s t u hst htu := Set.Subset.trans hst htu
+  le_antisymm s t hst hts := by
+    apply IrreducibleCloseds.ext
+    exact Set.Subset.antisymm hst hts
```

Comment:
Again, all this stuff not actually about Krull dimension needs to be in earlier files.